Homework-2/script.js

```diff
@@ -0,0 +1,36 @@
+  //Exercitiul 2
+const stopWatch = document.querySelector('#stopwatch');
+const startButton = document.querySelector('#start-stopwatch');
+const resetButton = document.querySelector('#reset-stopwatch');
+let timer;
+let millisecond = 0;
+function startWatch() {
+  clearInterval(timer);
+  timer = setInterval(()=> {
+    if (stopWatch.innerHTML !== '59 : 59 : 59') {
+      millisecond += 10;
+      let date = new Date(millisecond);
+      let milliseconds = String(date.getMilliseconds()).split(0,1);
+      let seconds = String(date.getSeconds()).padStart(2, "0");
+      let minutes = String(date.getMinutes()).padStart(2, "0");
+      stopWatch.innerHTML = `${minutes} : ${seconds} : ${milliseconds}`;
+    } else {
+        resetWatch();
+    }
+  }, 10);
+  startButton.classList.toggle('pause-btn');
+  if (startButton.classList.contains('pause-btn')) {
+    startButton.textContent = 'stop';
+  } else {
+    startButton.textContent = 'play';
+    clearInterval(timer);
+  }
+}
+function resetWatch() {
+  clearInterval(timer);
+  millisecond = 0;
+  startButton.textContent = 'play';
+  stopWatch.innerHTML = '00 : 00 : 00';
+}
+startButton.addEventListener('click', startWatch);
+resetButton.addEventListener('click', resetWatch);
```

Comment:
O abordare interesanta a exercitiului, care functioneaza corect, dar care s-ar putea sa fie cam "scump" daca e sa ne gandim la timpul de executie. Daca doresti, ai putea sa imbunatatesti doua aspecte ale acestei implementari: 1. afisarea milisecundelor pana la 10 sa aiba concatenat si un '0' in fata ca sa nu para ca sacadeaza cronometrul, iar 2. daca apesi butonul de `Reset` in timp ce cronometrul merge, textul butonului `Start`/`Stop` se va schimba, dar nu va functiona la primul click.